junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/Arguments.java

```diff
@@ -38,13 +38,47 @@
 	 */
 	Object[] get();
 
+	// todo: document properly this and below.
+	// todo: The fact there used to be `hasDescription` indicates we might be better of with Optional<String>
+	default String description() {
+		return "";
+	}
+
+// todo: do we need this?
+//	/** Returns true if these arguments have a non-empty description. **/
+//	default boolean hasDescription() {
+//		return !description().isEmpty();
+//	}
+
+	// todo: @MustCheckReturnValue
+	default Arguments description(String testCaseDescription) {
+		Preconditions.notBlank(testCaseDescription,
+				() -> "Test case description must not be blank: '" + testCaseDescription + "'!");
+
+		Object[] arguments = get();
```

Comment:
I don't think we should call `get()` here. We could return a decorator instead.

Reply:
May I ask why? Do you think it would be simpler, more readable, faster? `get()` simply returns a reference to the underlying array.

Reply:
See above comment.